build-tools/packages/build-tools/src/fluidBuild/tasks/task.ts

```diff
@@ -65,25 +65,47 @@ export abstract class Task {
 		// This function should only be called by task with task names
 		assert.notStrictEqual(this.taskName, undefined);
 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		this.dependentTasks = this.node.getDependentTasks(this, this.taskName!, pendingInitDep);
+		this.dependentTasks = this.node.getDependsOnTasks(this, this.taskName!, pendingInitDep);
 	}
 
 	protected get transitiveDependentLeafTask() {
-		if (this._transitiveDependentLeafTasks === undefined) {
-			const dependentTasks = this.dependentTasks;
-			if (dependentTasks) {
-				const s = new Set<LeafTask>();
-				for (const dependentTask of dependentTasks) {
-					dependentTask.transitiveDependentLeafTask.forEach((t) => s.add(t));
-					dependentTask.collectLeafTasks(s);
+		if (this._transitiveDependentLeafTasks === null) {
```

Comment:
Ahhh, are you using null to represent a unique state?

Reply:
Yes.  where the function is call, but hasn't completed.  use to detect cirulcar reference.